assets/app/view/game/auto.rb

```diff
@@ -39,10 +39,10 @@ def render_content
           }.freeze
 
           if !(available = @game.available_programmed_actions).empty?
-            enabled = @game.programmed_actions[sender]
             available.each do |type|
+              enabled = @game.programmed_actions[sender].find { |a| a.is_a?(type) }
               if (method = types[type])
```

Comment:
Since you are already refactoring, I think you can do something like "next unless (method = types[type])" at the top of the block.

Reply:
Yep.